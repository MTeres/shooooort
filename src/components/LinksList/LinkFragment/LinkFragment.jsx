import React from 'react'
import PropTypes from 'prop-types'
import styled, { css, keyframes } from 'styled-components'
import colors from '../../../styles/color'
import { CopyToClipboard } from 'react-copy-to-clipboard'
import TimeAgo from 'time-ago'

export default class LinkFragment extends React.Component {
  constructor () {
    super()
    this.state = {
      hover: false
    }
  }
  static propTypes = {
    link: PropTypes.object.isRequired
  }
  render () {
    const { link } = this.props
    const { hover } = this.state
    return (
      <Main>
        <CopyToClipboard text={`https://impraise-shorty.herokuapp.com/${link.shortcode}`}>
          <LinkWrapper hover={hover} onMouseEnter={() => this.setState({ hover: true })} onMouseLeave={() => this.setState({ hover: false })}>
            <Link>
              <h1>shooooort.com/
                <span>{link.shortcode}</span>
                {hover &&
                <CopyMessage>Click to copy</CopyMessage>
                }
              </h1>
            </Link>
            <h3>{link.url}</h3>
          </LinkWrapper>
        </CopyToClipboard>
        <RedirectCount isLoading={link.isUpdating}>
          {link.redirectCount}
        </RedirectCount>
        <Last isLoading={link.isUpdating}>
          {link.lastSeenDate ? TimeAgo.ago(new Date(link.lastSeenDate)) : 'Never visited'}
        </Last>
      </Main>
    )
  }
}

const Main = styled.div`
  display: flex;
  justify-content: space-between;
`

const LinkWrapper = styled.div`
  cursor: pointer;
  display: flex;
  flex-direction: column;
  font-weight: 400;
  overflow: hidden;
  width: 55%;
  color: ${colors.secondaryText};
   h3 {
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
   }
`

const Link = styled.div`
  h1 {
    color: ${colors.primaryText};
    display: flex;
    font-size: 16px;
    font-weight: 400;
  }
  span {
    color: ${colors.accent};
  }
`

const CopyMessage = styled.span`
  color: ${colors.accent};
  margin-left: auto;
  margin-right: 10px;
`

const RedirectCount = styled.div`
  width: 41px;
  height: 16px;
  text-align: center;
  ${p => p.isLoading && css`
    background-color: ${colors.primaryText};
    animation: ${LoadingAnimation} 1s infinite;
  `}
`

const Last = styled.div`
  width: 85px;
  min-height: 16px;
  height: fit-content;
  text-align: center;
  ${p => p.isLoading && css`
    background-color: ${colors.primaryText};
    animation: ${LoadingAnimation} 1s infinite;
  `}
`

const LoadingAnimation = keyframes`
  0% {
    transform: scale(.9);
  }
  100% {
    transform: scale(1.2);
  }
`
