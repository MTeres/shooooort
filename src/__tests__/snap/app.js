import React from 'react';
import LinkFragment from '../../components/LinksList/LinkFragment/LinkFragment'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

it('renders item correct', () => {
  const mockLink = {
    shortcode: 'oV9VC1',
    url: 'google.com',
    isUpdating: false,
    startDate: '2018-05-10T20:25:12+00:00',
    redirectCount: 0
  }
  const tree = renderer
    .create(<LinkFragment link={mockLink} />)
    .toJSON()
  expect(tree).toMatchSnapshot('link-with-0-visits')
})

// OTHER COMPONENTS HERE
