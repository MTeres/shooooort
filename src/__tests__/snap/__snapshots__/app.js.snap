// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders item correct: link-with-0-visits 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: 400;
  overflow: hidden;
  width: 55%;
  color: #AAA;
}

.c1 h3 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c2 h1 {
  color: #555;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: 400;
}

.c2 span {
  color: #EB4A42;
}

.c3 {
  width: 41px;
  height: 16px;
  text-align: center;
}

.c4 {
  width: 85px;
  min-height: 16px;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  text-align: center;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="c2"
    >
      <h1>
        shooooort.com/
        <span>
          oV9VC1
        </span>
      </h1>
    </div>
    <h3>
      google.com
    </h3>
  </div>
  <div
    className="c3"
  >
    0
  </div>
  <div
    className="c4"
  >
    Never visited
  </div>
</div>
`;
